### ISSUE TITLE
what error 'fun emqx_web_hook:on_client_connack/3' means ?

### ISSUE THREAD

**Issue by seqwait** [emqx/emqx#6619]
i hava a emqx swarm cluster node1 and node2 , they are a lot of error log just like below! and i don't know what's going on and how to resolve it.

docker version: Docker version 20.10.11, build dea9396
system: CentOS Linux release 8.4.2105
emqx version: emqx/emqx:v4.0.0

```
2022-01-04 06:50:53.389 [error] <<"android_27205_185063">>@10.0.0.8:3344 [Hooks] Failed to execute {fun emqx_web_hook:on_client_connack/3,
                           [{undefined}]}([#{clean_start => false,
                                             clientid =>
                                                 <<"android_27205_185063">>,
                                             conn_mod => emqx_connection,
                                             conn_props => undefined,
                                             expiry_interval => 7200,
                                             keepalive => 60,peercert => [],
                                             peername => {{10,0,0,8},3344},
                                             proto_name => <<"MQTT">>,
                                             proto_ver => 4,
                                             receive_maximum => 1024,
                                             sockname => {{10,0,0,72},1883},
                                             socktype => proxy,
                                             username => <<"27205">>},
                                           success,#{}]): {{badarity,
                                                            {fun emqx_web_hook:on_client_connack/3,
                                                             [#{clean_start =>
                                                                 false,
                                                                clientid =>
                                                                 <<"android_27205_185063">>,
                                                                conn_mod =>
                                                                 emqx_connection,
                                                                conn_props =>
                                                                 undefined,
                                                                expiry_interval =>
                                                                 7200,
                                                                keepalive =>
                                                                 60,
                                                                peercert => [],
                                                                peername =>
                                                                 {{10,0,0,8},
                                                                  3344},
                                                                proto_name =>
                                                                 <<"MQTT">>,
                                                                proto_ver => 4,
                                                                receive_maximum =>
                                                                 1024,
                                                                sockname =>
                                                                 {{10,0,0,72},
                                                                  1883},
                                                                socktype =>
                                                                 proxy,
                                                                username =>
                                                                 <<"27205">>},
                                                              success,#{},
                                                              {undefined}]}},
                                                           [{emqx_hooks,
                                                             execute,2,
                                                             [{file,
                                                               "/emqx_rel/_build/emqx/lib/emqx/src/emqx_hooks.erl"},
                                                              {line,176}]},
                                                            {emqx_hooks,
                                                             safe_execute,2,
                                                             [{file,
                                                               "/emqx_rel/_build/emqx/lib/emqx/src/emqx_hooks.erl"},
                                                              {line,164}]},
                                                            {emqx_hooks,
                                                             do_run_fold,3,
                                                             [{file,
                                                               "/emqx_rel/_build/emqx/lib/emqx/src/emqx_hooks.erl"},
                                                              {line,143}]},
                                                            {emqx_channel,
                                                             handle_out,3,
                                                             [{file,
                                                               "/emqx_rel/_build/emqx/lib/emqx/src/emqx_channel.erl"},
                                                              {line,1305}]},
                                                            {emqx_connection,
                                                             with_channel,3,
                                                             [{file,
                                                               "/emqx_rel/_build/emqx/lib/emqx/src/emqx_connection.erl"},
                                                              {line,537}]},
                                                            {emqx_connection,
                                                             process_msg,3,
                                                             [{file,
                                                               "/emqx_rel/_build/emqx/lib/emqx/src/emqx_connection.erl"},
                                                              {line,284}]},
                                                            {proc_lib,
                                                             wake_up,3,
                                                             [{file,
                                                               "proc_lib.erl"},
                                                              {line,259}]}]}
```

**Comment by zmstone** [comment]
@seqwait it's a bug in the callback
client.connack hook point should accept 3 args but implemented with 4

**Comment by seqwait** [comment]
> client.connack hook point should accept 3 args but implemented with 4

does it affect normal function? and how to fixed it! @zmstone

**Comment by zmstone** [comment]
it seems only affecting this hook point.
We'll fix it in the next release `4.3.12`
